sign/bdn/mask.go

```diff
@@ -170,3 +202,14 @@ func (m *Mask) Merge(mask []byte) error {
 
 	return nil
 }
+
+// Clone copies the mask while keeping the precomputed coefficients, etc. This method is thread safe
+// and does not modify the original mask. Modifications to the new Mask will not affect the original.
+func (m *Mask) Clone() *Mask {
+	return &Mask{
+		mask:        slices.Clone(m.mask),
+		publics:     m.publics,
+		publicCoefs: m.publicCoefs,
+		publicTerms: m.publicTerms,
```

Comment:
The comment is not entirely true for the fields other than `mask`, no? But I guess since these are non-exported and we only read from them in the `bdn` code that should be fine.

Reply:
Exactly. The other fields are non-exported and immutable. I've documented that.